### ISSUE TITLE
[1.12] Backpack item has a missing texture.

### ISSUE THREAD

**Issue by elucent** [copygirl/WearableBackpacks#42]
Using Wearable Backpacks 1.12-3.0.3.

**Comment by copygirl** [comment]
Someone reported this on CurseForge so I tried reproducing this with the latest Forge version yesterday, and could not. Is this an issue with a particular Forge version or another mod?

Please see if you run into the same issue if you use latest Forge, and if so, if it only happens with Wearable Backpacks enabled. If everything works fine then, it must be a conflict with another mod. I would appreciate it if you could find out which one that is.

**Comment by PaleoCrafter** [comment]
This might be related to [VanillaImmersion#12](https://github.com/PaleoCrafter/VanillaImmersion/issues/12). I'm currently looking into it, but it seems to be related to `Item.delegate` being messed up for some reason. I'll report back when I have clear results.

**Comment by PaleoCrafter** [comment]
I've found the culprit: You set the item model during the pre-init phase (see [here](https://github.com/copygirl/WearableBackpacks/blob/master/src/main/java/net/mcft/copy/backpacks/ProxyClient.java#L70). At this point, the `Item` hasn't been registered and hence `Item.delegate` does not have a valid name. Unfortunately, `ModelLoader` uses the delegate's name as part of the key for MRLs. I made the same mistake in my mod and hence my hammer had the model of your backpack while yours got a missing model. I assume that you were unable to reproduce this because Vanilla does not use the `ModelLoader` functions.

The fix for this issue is to use `ModelRegistryEvent` and to have all calls to `setCustomModelResourceLocation` there.

Whatever caused the issue in @elucent's setup probably has to change this as well.
